OpenRA.Game/Traits/Selectable.cs

```diff
@@ -9,13 +9,29 @@
  */
 #endregion
 
+using System;
+
 namespace OpenRA.Traits
 {
-	[Desc("This actor is selectable. Defines bounds of selectable area, selection class and selection priority.")]
+	[Flags]
+	public enum SelectionPriorityModes
+	{
+		None = 0,
+		FlatPriority = 1,
+		LowPriority = 2
+	}
+
+	[Desc("This actor is selectable. Defines bounds of selectable area, selection class and selection priority. Defines whether actor is included in selection modes.")]
 	public class SelectableInfo : InteractableInfo
 	{
 		public readonly int Priority = 10;
 
+		[Desc("Allow selection priority to be modified using a hotkey.",
+			"Valid values are None (priority is not affected by modifiers)",
+			"FlatPriority (priority is set to 10 when Ctrl pressed) and",
+			"LowPriority (priority of actors without this value is set to 0 when Alt pressed).")]
+		public readonly SelectionPriorityModes SelectionPriorityModes = SelectionPriorityModes.FlatPriority;
```

Comment:
I might be missing something here, but doesn't FlatPriority change the default behavior compared to bleed? Shouldn't this be None instead?

Reply:
Either of the values here changes the default behavior when a modifier key (<kbd>Ctrl</kbd> or <kbd>Alt</kbd>) is pressed. Even if it's `None` the priority would be set to `0` when <kbd>Ctrl</kbd> is pressed during a selection. 

Perhaps the description here is not accurate. Actually I've been struggling to come up with good names and description for this feature. Because even though the modes are called "flat" and "low priority" they are not tied to the priority of an actor. They may as well be called "select ctrl" and "select alt"...

Reply:
A value of `None` probably makes more sense as a default. I didn't do it like that because most of the actors are included in the "flat" mode and I wanted to spare some YAML lines...

Reply:
IMO the difficulty in describing this probably reflects that this isn't exposed to the yaml in the best way.

What if we changed the behaviour such that `None`, the default, keeps things as they are on bleed. Then expose usefully named templates that can be inherited, e.g. `Inherits@SELECTIONPRIORITY: ^SupportActorSelectionPriority`?

Reply:
I agree that it's probably not the best way to expose it in YAML. I am open to suggestions. However I don't understand the template and inheritance idea -- could you provide an example or point me to a similar implementation of another feature?

Reply:
The `^AutoTarget*` templates that most actors implement.